app/Http/Controllers/TasksController.php

```diff
@@ -67,65 +67,45 @@ public function show($id)
     /**
      * Show the form for editing the specified resource.
      *
-     * @param  int  $id
+     * @param \App\Models\Task $task
      * @return \Illuminate\Http\Response
      */
-    public function edit($id)
+    public function edit(Task $task)
     {
-        $task = Task::find($id);
-
         if (empty($task)) return response(['message' => 'Task not found'], 404);
 
-        return view('templates.edit-task', ['task' => $task]);
+        return view('templates.edit-task', $task);
```

Comment:
Just a small problem found in that part of the code. When trying to edit the task, it returned undefined $task, so I needed to add the method compact('task')  passing task and it worked.

would be like this: return view('templates.edit-task', compact('task'));

Reply:
Oh sorry, will update it.